sky/data/data_utils.py

```diff
@@ -225,3 +329,120 @@ def run_upload_cli(command: str, access_denied_message: str, bucket_name: str):
                 raise exceptions.StorageUploadError(
                     f'Upload to bucket failed for store {bucket_name}. '
                     'Please check the logs.')
+
+
+def store_rclone_config(bucket_name: str,
+                        region: str,
+                        apply_changes: bool = True):
+    """creates a configuration files for rclone - used for
+    bucket syncing and mounting """
+
+    rclone_config_path = os.path.expanduser(RCLONE_CONFIG_PATH)
+    access_key_id, secret_access_key = ibm.get_hmac_keys()
+
+    config_data = textwrap.dedent(f"""\
+        [{bucket_name}]
+        type = s3
+        provider = IBMCOS
+        access_key_id = {access_key_id}
+        secret_access_key = {secret_access_key}
+        region = {region}
+        endpoint = s3.{region}.cloud-object-storage.appdomain.cloud
+        location_constraint = {region}-smart
+        acl = private
+        """)
+
+    # create ~/.config/rclone/ if doesn't exist
+    os.makedirs(os.path.dirname(rclone_config_path), exist_ok=True)
+    # create rclone.conf if doesn't exist
+    if not os.path.isfile(rclone_config_path):
+        open('file', 'w').close()
+
+    with FileLock(rclone_config_path + '.lock'):
+        profiles_to_keep = _remove_bucket_profile_rclone(bucket_name)
+
+        # write back file without profile: [bucket_name]
+        # to which the new bucket profile is appended
+        if apply_changes:
+            with open(f'{rclone_config_path}', 'w') as file:
+                if profiles_to_keep:
+                    file.writelines(profiles_to_keep)
```

Comment:
The file will be opened regardless of `if profiles_to_keep`, since `config_data` is to be written.
I do agree however that the file lock should only be set if `apply_changes`.